packages/spectacular-docs/docs/feature-testing/angular-testing-library.md

```diff
@@ -0,0 +1,575 @@
+});
+```
+
+[`SpectacularAppComponent`](../api/classes/spectacularappcomponent) is a simple
+component with a router outlet which will host all our feature's routes.
+
+Notice that we resolve the render result and store it in a variable of type
+[`RenderResult<SpectacularAppComponent, SpectacularAppComponent>`](https://testing-library.com/docs/angular-testing-library/api#renderresult):
+
+```ts {24}
+import {
+  SpectacularAppComponent,
+  SpectacularFeatureTestingModule,
+} from '@ngworker/spectacular';
+import { render, RenderResult } from '@testing-library/angular';
+import {
+  CrisisCenterModule,
+  crisisCenterPath,
+} from '@tour-of-heroes/crisis-center';
+
+describe('Tour of Heroes: Crisis center integration tests', () => {
+  beforeEach(async () => {
+    result = await render(SpectacularAppComponent, {
+      excludeComponentDeclaration: true,
+      imports: [
+        SpectacularFeatureTestingModule.withFeature({
+          featureModule: CrisisCenterModule,
+          featurePath: crisisCenterPath,
+        }),
+      ],
+    });
+  });
+
+  let result: RenderResult<SpectacularAppComponent, SpectacularAppComponent>;
+});
+```
+
+Next, we register our feature by calling
+[`SpectacularFeatureTestingModule.withFeature`](..//api/classes/spectacularfeaturetestingmodule#withfeature)
+and adding the result to the `imports` option:
+
+```ts {16-19}
+import {
+  SpectacularAppComponent,
+  SpectacularFeatureTestingModule,
+} from '@ngworker/spectacular';
+import { render, RenderResult } from '@testing-library/angular';
+import {
+  CrisisCenterModule,
+  crisisCenterPath,
+} from '@tour-of-heroes/crisis-center';
+
+describe('Tour of Heroes: Crisis center integration tests', () => {
+  beforeEach(async () => {
+    result = await render(SpectacularAppComponent, {
+      excludeComponentDeclaration: true,
+      imports: [
+        SpectacularFeatureTestingModule.withFeature({
+          featureModule: CrisisCenterModule,
+          featurePath: crisisCenterPath,
+        }),
+      ],
+    });
+  });
+
+  let result: RenderResult<SpectacularAppComponent, SpectacularAppComponent>;
+});
+```
+
+We pass our routed feature module to the
+[`featureModule`](../api/interfaces/spectacularfeaturetestingmoduleoptions#featuremodule)
+option, in our case `CrisisCenterModule`.
+
+Additionally, we pass the route path that our feature module uses in our
+application to the
+[`featurePath`](../api/interfaces/spectacularfeaturetestingmoduleoptions#featurepath)
+option, in our case the value of the `crisisCenterPath` variable which is used
+between our application and our feature test suite.
+
+The [`SpectacularAppComponent`](../api/classes/spectacularappcomponent) is
+declared by
+[`SpectacularFeatureTestingModule.withFeature`](..//api/classes/spectacularfeaturetestingmodule#withfeature)
+so we set the
+[`excludeComponentDeclaration`](https://testing-library.com/docs/angular-testing-library/api#excludecomponentdeclaration)
```

Comment:
So this would be run in the main test file of the project? Because the benefit of a feature test is that we can reuse the same setup for a lot of test cases.

Reply:
Yep, correct. See https://github.com/testing-library/angular-testing-library/blob/main/apps/example-app/src/test-setup.ts for an example. It would only be useful when all tests are using that config.

Reply:
Thanks, @timdeschryver.

I noticed that the interface of Angular Testing Library's `configure` options object is not listed in the API reference.